modules/wr2_ethlovatoaevu/main.sh

```diff
@@ -1,10 +1,28 @@
 #!/bin/bash
+OPENWBBASEDIR=$(cd `dirname $0`/../../ && pwd)
+RAMDISKDIR="${OPENWBBASEDIR}/ramdisk"
+MODULEDIR=$(cd `dirname $0` && pwd)
+DMOD="PV"
+#DMOD="MAIN"
+Debug=$debug
 
-if (( pv2kitversion == 1 )); then
-	python /var/www/html/openWB/modules/wr2_ethlovatoaevu/readsdm.py  
+#For development only
+#Debug=1
+
+if [ ${DMOD} == "MAIN" ]; then
+	MYLOGFILE="${RAMDISKDIR}/openWB.log"
 else
-	python /var/www/html/openWB/modules/wr2_ethlovatoaevu/readlovato.py  
+	MYLOGFILE="${RAMDISKDIR}/nurpv.log"
 fi
 
-pvwatt2=$(</var/www/html/openWB/ramdisk/pv2watt)
-echo $pvwatt2
+#python3 ${OPENWBBASEDIR}/modules/wr_pvkitflex/test.py "2" ${pvflexip} ${pvflexport} ${pvflexid} >>${MYLOGFILE} 2>&1
+
+if (( pvkitversion == 1 )); then
+	python3 ${OPENWBBASEDIR}/modules/wr_pvkit/readlovato.py "2" "192.168.193.15" "8899" "0x08" >>${MYLOGFILE} 2>&1
```

Comment:
Funktioniert das? Hat das mal funktioniert?

Hintergrund der Frage: Während ich so durch den Code sehe fällt mir auf, dass einige Python Dateien die Methode `decode` von `str` aufrufen. Die Methode [gab es in Python2](https://docs.python.org/2.7/library/stdtypes.html?highlight=decode#str.decode). Ich habe es nicht so eindeutig irgendwo stehen sehen, aber offenbar gibt es die Methode in Python3 nicht.

Bis zu diesem PR wurde die Datei `/modules/wr_pvkit/readlovato.py` mit python2 gestartet. Ab hier ist es python3. Diese `decode`-Methode wird in der `readlovato.py` aber ganz viel genutzt.

Das würde mich zu der Vermutung führen, dass das ganze Skript nicht funktioniert?!

Reply:
Da hast Du Recht. Bei Strings kann (und muss) das .decode ersatzlos gestrichen werden.

Reply:
Da die geänderten Module innerhalb einer Woche durch neue ersetzt wurden, sind da sicher Dinge gar nicht hoch gekommen.